features/sketch/src/main/java/com/wire/android/feature/sketch/DrawingCanvasBottomSheet.kt

```diff
@@ -17,70 +17,209 @@
+        },
+        sheetState = sheetState,
+        onDismissRequest = onDismissSketch
+    ) {
+        Column(modifier = Modifier.fillMaxSize()) {
+            Row(Modifier.weight(weight = 1f, fill = true)) {
+                DrawingCanvasComponent(
+                    state = viewModel.state,
+                    onStartDrawingEvent = viewModel::onStartDrawingEvent,
+                    onDrawEvent = viewModel::onDrawEvent,
+                    onStopDrawingEvent = viewModel::onStopDrawingEvent,
+                    onSizeChanged = viewModel::onSizeChanged,
+                    onStartDrawing = viewModel::onStartDrawing,
+                    onDraw = viewModel::onDraw,
+                    onStopDrawing = viewModel::onStopDrawing
+                )
+            }
             Row(
                 Modifier
-                    .fillMaxWidth(),
-                horizontalArrangement = Arrangement.SpaceBetween
+                    .height(dimensions().spacing80x)
+                    .fillMaxWidth()
             ) {
-                val scope = rememberCoroutineScope()
-                IconButton(
-                    onClick = {
-                        scope.launch { sheetState.hide() }.invokeOnCompletion { onDismissSketch() }
-                    },
-                ) {
-                    Icon(
-                        Icons.Filled.Close,
-                        contentDescription = stringResource(
-                            com.google.android.material.R.string.mtrl_picker_cancel
-                        )
-                    )
-                }
-                IconButton(
-                    onClick = {
-                        onSendSketch()
-                    },
-                ) {
-                    Icon(
-                        Icons.Filled.Send,
-                        contentDescription = stringResource(
-                            com.google.android.material.R.string.mtrl_picker_cancel
-                        )
-                    )
-                }
-
+                DrawingToolbar(
+                    state = viewModel.state,
+                    onSendSketch = {
+                        scope.launch { onSendSketch(viewModel.saveImage(context, tempWritableImageUri)) }
+                            .invokeOnCompletion { scope.launch { sheetState.hide() } }
+                    }
+                )
             }
-        },
-        sheetState = sheetState,
-        onDismissRequest = {
-            onDismissSketch()
         }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun DrawingTopBar(
+    scope: CoroutineScope = rememberCoroutineScope(),
+    sheetState: SheetState,
+    conversationTitle: String,
+    onDismissSketch: () -> Unit,
+    onUndoStroke: () -> Unit,
+    state: DrawingState
+) {
+    Row(
+        Modifier
+            .fillMaxWidth()
+            .padding(horizontal = dimensions().spacing8x)
+    ) {
+        Row(
```

Comment:
Do you need 2 rows?

Reply:
Good catch, all of these, were leftovers from an old approach I was trying to get the bottom sheet not draggable.